src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.asm

```diff
@@ -317,4 +319,74 @@ ExchangeRetry
 
     LEAF_END RhpCheckedXchg
 
+;; RhpAssignRefArm64(Object** dst, Object* src)
+;;
+;; On entry:
+;;   x14  : the destination address (LHS of the assignment).
+;;   x15  : the object reference (RHS of the assignment).
+;;
+;; On exit:
+;;   x9   : trashed
+;;   x10  : trashed if WRITE_BARRIER_CHECK
+;;   x14  : trashed
+;;
+    LEAF_ENTRY RhpAssignRefArm64, _TEXT
+    ALTERNATE_ENTRY RhpAssignRefAVLocation
+    ALTERNATE_ENTRY RhpAssignRefX1AVLocation
+
+        stlr    x15, [x14]
+
+        INSERT_UNCHECKED_WRITE_BARRIER_CORE x14, x15
+
+        ret
+    LEAF_END RhpAssignRefArm64
+
+;; void JIT_CheckedWriteBarrier(Object** dst, Object* src)
+;; On entry:
+;;   x14 : the destination address (LHS of the assignment)
+;;   x15 : the object reference (RHS of the assignment)
+;;
+;; On exit:
+;;   x9  : trashed
+;;   x10 : trashed if WRITE_BARRIER_CHECK
+;;   x14 : incremented by 8 to implement JIT_ByRefWriteBarrier contract
+;;
+    LEAF_ENTRY RhpCheckedAssignRefArm64, _TEXT
+    ALTERNATE_ENTRY RhpCheckedAssignRefAVLocation
+    ALTERNATE_ENTRY RhpCheckedAssignRefX1AVLocation
+
+        stlr    x15, [x14]
+
+        ;; TODO: This trashes x14
+        INSERT_CHECKED_WRITE_BARRIER_CORE x14, x15
+
+        add x14, x14, #8
+
+        ret
+    LEAF_END RhpCheckedAssignRefArm64
+
+;; void JIT_ByRefWriteBarrier
+;; On entry:
+;;   x13 : the source address (points to object reference to write)
+;;   x14 : the destination address (object reference written here)
+;;
+;; On exit:
+;;   x9  : trashed
+;;   x10 : trashed if WRITE_BARRIER_CHECK
+;;   x13 : incremented by 8
+;;   x14 : incremented by 8
+;;
+    LEAF_ENTRY RhpByRefAssignRefArm64, _TEXT
+        ldr x15, [x13]
+        str x15, [x14]
```

Comment:
Does this need to be `stlr` ?

Reply:
Probably yes, however, I see no `stlr`/`dmb` in Linux ARM64 and LinuxARM32 versions:
https://github.com/dotnet/corert/commit/39c6ceb72d3764e48504ac26ec926e7b648c4752#diff-5f1553cb1c44e19848424918ad7a351a00fce13fd08216e3d866cc41f1e0bf0fR360
https://github.com/dotnet/corert/pull/4235/commits/e08c6b80f650f280414b22843125ccde07b1c29e#diff-99ef768b3a97ff13d34b55ad20908bb94982d34ff51f8b2a91efbd4dc010e6c0R289

Reply:
It may be a bug

Reply:
I will use `stlr` for Windows ARM64.  I filed #639 for Linux ARM32 and ARM64.